ucte/ucte-network/src/main/java/com/powsybl/ucte/network/UcteNodeCode.java

```diff
@@ -91,6 +96,32 @@ public void setBusbar(Character busbar) {
         this.busbar = busbar;
     }
 
+    /**
+     * Convert a string into a UcteNodeCode if possible, i.e. the string is compatible with the ucteNodeCode format
+     * @param id to convert into a UcteNodeCode
+     * @return an Optional that may contain a UcteNodeCode
+     */
+    //TODO : Need to find a way to handle non UCTE ids
+    public static Optional<UcteNodeCode> parseUcteNodeCode(String id) {
```

Comment:
When I created this method the goal was following: if the string can be converted in UcteNodeCode because it's ucte compliant, do it and return it. It's not the goal of this method to create a UcteNodeCode from a String that is not already ucte compliant. We let the handling of the case where it's not a ucte compliant id to other methods.